libagent/device/ui/pinentry_gpg.py

```diff
@@ -0,0 +1,51 @@
+"""Python wrapper for GnuPG's pinentry."""
+import os
+import subprocess
+import sys
+
+import libagent.gpg.agent
+
+
+def write(p, line):
+    """Send and flush a single line to the subprocess' stdin."""
+    p.stdin.write(line)
+    p.stdin.flush()
+
+
+def readline(p):
+    """Read a single line from the subprocess' stdout."""
+    return p.stdout.readline()
+
+
+def expect(p, prefixes):
+    """Read a line and return it without required prefix."""
+    resp = readline(p)
+    for prefix in prefixes:
+        if resp.startswith(prefix):
+            return resp[len(prefix):]
+    raise ValueError('Unexpected response: {}'.format(resp))
+
+
+def main():
+    """Use GPG pinentry program to interact with the user."""
+    args = ['pinentry']
```

Comment:
Could we make this configurable?

With the module name it's a bit more complex since it adds a second layer of configuration.  It might need to be an absolute path too in some circumstances since PATH is empty in cron, systemd, etc.

Reply:
You're right...
I have added an option to pass arguments to configure the actual binary that is executed by `pinetry_gpg.py` at https://github.com/romanz/trezor-agent/commit/1a50d0a03570f5f4ddbb8d00b22bfa56d7bd9a6f.